src/algebra/homology/homological_complex.lean

```diff
@@ -74,6 +74,22 @@ begin
   { rw [C.shape i j hij, zero_comp] }
 end
 
+@[ext]
+lemma ext {C₁ C₂ : homological_complex V c} (h_X : C₁.X = C₂.X)
+  (h_d : ∀ (i j : ι), c.rel i j → C₁.d i j ≫ eq_to_hom (congr_fun h_X j) =
+    eq_to_hom (congr_fun h_X i) ≫ C₂.d i j) : C₁ = C₂ :=
```

Comment:
This lemma seems like a bad idea, inviting people to argue in a very evil manner. :-) What do you actually want this for?

Reply:
It is used in the above `map_of`, which I believe is not evil! (I understand the concern, but in general, I would not agree that equalities between two objects are always evil. If we only provide isomorphisms instead of equalities, then if we have three equal objects, we would sometimes to have check transitivity of the isomorphisms, which we get for free using `eq_to_hom_trans`.)

Reply:
Sorry I haven't had time to think about this more. Perhaps let's remove the `@[ext]` for now, and apply it by hand? I fear this could accidentally introduce more unpleasant goals than people are expecting.

Reply:
Thanks for the review. I have removed the `ext` attribute.